compiler/semdata.nim

```diff
@@ -501,18 +501,10 @@ proc errorNode*(c: PContext, n: PNode): PNode =
   result = newNodeI(nkEmpty, n.info)
   result.typ = errorType(c)
 
-proc localErrorNode*(c: PContext, n: PNode, info: TLineInfo, msg: TMsgKind, arg: string): PNode =
-  localError(c.config, info, msg, arg)
-  result = errorNode(c, n)
-
-proc localErrorNode*(c: PContext, n: PNode, info: TLineInfo, arg: string): PNode =
-  localErrorNode(c, n, info, errGenerated, arg)
-
-proc localErrorNode*(c: PContext, n: PNode, msg: TMsgKind, arg: string): PNode =
-  localErrorNode(c, n, n.info, msg, arg)
-
-proc localErrorNode*(c: PContext, n: PNode, arg: string): PNode =
-  localErrorNode(c, n, n.info, errGenerated, arg)
+template localErrorNode*(c: PContext, n: PNode, arg: string, info2: auto = (), kind: TMsgKind = errGenerated): PNode =
+  let info = when info2 is tuple[]: n.info else: info2
```

Comment:
This is a hack, please split it into the overloads again.

Reply:
done, i added back 1 overload to unblock this PR, but IMO this is a useful pattern that can be reused in similar contexts, I don't see how it's a hack